plugins/invisible-typing/src/index.ts

```diff
@@ -12,7 +12,7 @@ export interface SettingsType {
   button?: boolean;
   invisible?: boolean;
   channelWise?: boolean;
-  channels?: Record<string, boolean>;
+  channels?: Record<string, boolean | undefined>;
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (code_refinement):** Adjustment to the 'channels' type to allow undefined values aligns with runtime behavior.

This change accurately reflects that channel settings might not be defined for every channel, allowing for more precise type checking and potentially catching errors at compile time.

```suggestion
channels?: Partial<Record<string, boolean>>;
```